core/jobs/batch_jobs/exp_tags_validation_jobs.py

```diff
@@ -0,0 +1,156 @@
+
+
+class GetNumberOfExpHavingInvalidTagsListJob(base_jobs.JobBase):
+    """Job that returns exploration having tags list length more than 10"""
+
+    def _check_tags_validity(self, tags: List[str]) -> bool:
+        """checks whether tags list is valid or not.
+
+        Args:
+            tags: List[str]. The list of all tag in exlporation.
+
+        Returns:
+            bool. If list contain invalid tag or tags length is more than 10
+            then it return true, otherwise false.
+        """
+        if len(tags) > 10:
+            return True
+        if len(set(tags)) < len(tags):
+            return True
+        for tag in tags:
+            if tag.strip() == '':
+                return True
+            elif len(tag) > 30:
+                return True
+        return False
+
+    def _get_description_of_wrong_exp(self, tags: List[str]) -> str:
+        """Returns the description of invalid tags property.
+
+        Args:
+            tags: List[str]. The list of all tag in exlporation.
+
+        Returns:
+            str. Description of invalid tags property.
+        """
+        output_string = ''
+        visited = set()
+        dup: List[str] = []
+        empty_tag = max_length_exceed_tag = 0
+
+        if len(tags) > 10:
+            output_string += ' tags length more than 10,'
+        if len(set(tags)) < len(tags):
+            for ele in tags:
+                if ele.strip() != '':
+                    if ele in visited or (visited.add(ele)):
+                        dup.append(ele)
+            if len(dup) != 0:
+                output_string += f' {len(dup)} duplicate values {dup},'
+
+        for tag in tags:
+            if tag.strip() == '':
+                empty_tag += 1
+            elif len(tag) > 30:
+                max_length_exceed_tag += 1
+
+        if (empty_tag != 0) or (max_length_exceed_tag != 0):
+            output_string += (
+                f' {empty_tag} empty tag and'
+                + f' {max_length_exceed_tag} tag having length more than 30, ')
+
+        last_comma_index = output_string.rfind(',')
+        output_string = (
+            output_string[:last_comma_index] + '.'
+            + output_string[last_comma_index + 1:])
+
+        return output_string.rstrip()
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns PCollection of invalid explorations with their id and
+        actual tags list.
+
+        Returns:
+            PCollection. Returns PCollection of invalid explorations with
+            their id and actual tags list.
+        """
+        total_explorations = (
+            self.pipeline
+            | 'Get all ExplorationModels' >> ndb_io.GetModels(
+                exp_models.ExplorationModel.get_all(include_deleted=False))
+            | 'Get exploration from model' >> beam.Map(
+                exp_fetchers.get_exploration_from_model)
+        )
+
+        exp_ids_with_exceeding_max_tags_len = (
```

Comment:
Consider changing it to 
```suggestion
        exp_ids_with_invalid_tags = (
```
because we are not only checking the tags list length but also checking its individual length, Just a suggestion.

Reply:
Yeah, you are right!!
Done!